plugins/destination/clickhouse/typeconv/ch/values/time.go

```diff
@@ -0,0 +1,40 @@
+package values
+
+import (
+	"time"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+)
+
+type date interface {
+	ToTime() time.Time
+}
+
+func dateValue[A date](arr primitive[A]) []*time.Time {
```

Comment:
what's the point of generics here? can't it be just an interface ?

Reply:
We're inferring the interface of the value returned by `Value(int)`. In this case, it's either `arrow.Date32` & `arrow.Date64`.
If we were to use interfaces there would've been 2 separate functions (we might as well just do 2 separate functions for each array type).